pkg/parser/task/task.go

```diff
@@ -56,34 +60,38 @@ func NewTask(
 		return nil
 	})
 
-	task.CollectibleField("container",
-		instance.NewCommunityContainer(environment.Merge(task.proto.Environment, env), boolevator).Schema(),
-		func(node *node.Node) error {
-			inst := instance.NewCommunityContainer(environment.Merge(task.proto.Environment, env), boolevator)
-			containerInstance, err := inst.Parse(node)
-			if err != nil {
-				return err
-			}
+	if _, ok := additionalInstances["container"]; !ok {
+		task.CollectibleField("container",
+			instance.NewCommunityContainer(environment.Merge(task.proto.Environment, env), boolevator).Schema(),
+			func(node *node.Node) error {
+				inst := instance.NewCommunityContainer(environment.Merge(task.proto.Environment, env), boolevator)
+				containerInstance, err := inst.Parse(node)
+				if err != nil {
+					return err
+				}
 
-			// Retrieve the platform to update the base environment
-			env["CIRRUS_OS"] = strings.ToLower(containerInstance.Platform.String())
+				// Retrieve the platform to update the base environment
+				env["CIRRUS_OS"] = strings.ToLower(containerInstance.Platform.String())
 
-			anyInstance, err := ptypes.MarshalAny(containerInstance)
-			if err != nil {
-				return err
-			}
-			task.proto.Instance = anyInstance
+				anyInstance, err := ptypes.MarshalAny(containerInstance)
+				if err != nil {
+					return err
+				}
+				task.proto.Instance = anyInstance
+				task.proto.Environment = environment.Merge(task.proto.Environment, map[string]string{"CIRRUS_OS": "linux"})
```

Comment:
How does this relate to the `env["CIRRUS_OS"]` above and why is it always `linux`?

Reply:
Right, can populate though `platform` field. See 445dbb7ba7f982fc57628affc02a988cdfa4eba4